File: app/src/main/java/catchcoins/box/Coin.java
package catchcoins.box;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;


import java.util.Random;

public class Coin {
    Bitmap[] coins = new Bitmap[4];
    int coinFrame = 0;
    int coinX, coinY, coinVelocity;
    Random random;

    public Coin (Context context) {
        coins[0] = BitmapFactory.decodeResource(context.getResources(), R.drawable.moneta1);
        coins[1] = BitmapFactory.decodeResource(context.getResources(), R.drawable.moneta2);
        coins[2] = BitmapFactory.decodeResource(context.getResources(), R.drawable.moneta3);
        coins[3] = BitmapFactory.decodeResource(context.getResources(), R.drawable.moneta4);
        random = new Random();
        resetPosition();
    }

    public void resetPosition() {
        coinX = random.nextInt(GameView.dWidth - getCoinWidth());
        coinY = -200+random.nextInt(600)*-1;
        coinVelocity = 35+random.nextInt(16);
    }

    public Bitmap getCoin(int coinFrame) {
        return coins[coinFrame];
    }
    public int getCoinWidth() {
        return coins[0].getWidth();
    }

    public int getCoinHeght() {
        return coins[0].getHeight();
    }
}

File: app/src/main/java/com/edurda77/catchcoins/Coin.kt
package com.edurda77.catchcoins

import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import java.util.Random

class Coin(context: Context) {
    var coins = arrayOf<Bitmap>()
    var coinFrame = 0
    var coinX = 0
    var coinY = 0
    var coinVelocity = 0
    var random: Random

    init {
        coins[1] = BitmapFactory.decodeResource(context.resources, R.drawable.moneta1)
        coins[2] = BitmapFactory.decodeResource(context.resources, R.drawable.moneta2)
        coins[3] = BitmapFactory.decodeResource(context.resources, R.drawable.moneta3)
        coins[4] = BitmapFactory.decodeResource(context.resources, R.drawable.moneta4)
        random = Random()
        resetPosition()
    }

    fun getCoin(coinFrame: Int): Bitmap {
        return coins[coinFrame]
    }

    val coinWidth: Int
        get() = coins[0].width
    val coinHeight: Int
        get() = coins[0].height

    fun resetPosition() {
        coinX = random.nextInt(GameView.dWidth - coinWidth)
        coinY = -200 + random.nextInt(600) * -1
        coinVelocity = 34 + random.nextInt(16)
    }
}